bindings/python/Jamfile

```diff
@@ -110,14 +112,18 @@ else
 		# debian jessie (1.55.0): libboost_python-py34.so.1.55.0
 		# boost Jamfile:          libboost_python38.so.1.73.0
 
-		local py-version = [ $(properties).get <python> ] ;
-
+		local py-version-str = [ $(properties).get <python> ] ;
+		local py-version = "" ;
 		local infix = "" ;
-		if $(py-version) < 3.7 && [ $(properties).get <target-os> ] = linux
-		{
-			infix = "-py" ;
+		if $(py-version-str) {
+			py-version = [ SPLIT_BY_CHARACTERS $(py-version-str) : "." ] ;
+
+			if [ version.version-less $(py-version) : 3 7 ] && [ $(properties).get <target-os> ] = linux
+			{
+				infix = "-py" ;
+			}
 		}
-		local boost-python-lib = "boost_python" $(infix) [ SPLIT_BY_CHARACTERS $(py-version) : "." ] ;
+		local boost-python-lib = "boost_python" $(infix) $(py-version-str) ;
```

Comment:
I think this should be:

```suggestion
		local boost-python-lib = "boost_python" $(infix) $(py-version) ;
```

Otherwise I get the following error:

```
/usr/bin/ld: cannot find -lboost_python3.10
```